piezo/PDXC2.py

```diff
@@ -0,0 +1,227 @@
+
+import clr
+import time
+import traceback
+from .GenericDevice import GenericDevice
+
+# Add references so Python can see .Net
+clr.AddReference("System")
+clr.AddReference("Thorlabs.MotionControl.Benchtop.PiezoCLI")
+clr.AddReference("Thorlabs.MotionControl.GenericPiezoCLI")
+clr.AddReference("Thorlabs.MotionControl.Controls")
+
+import System.Collections
+from System.Collections import *
+# Generic device manager
+import Thorlabs.MotionControl.Controls
+
+from Thorlabs.MotionControl.Benchtop.Piezo import *
+from Thorlabs.MotionControl.Benchtop.PiezoCLI import *
+from Thorlabs.MotionControl.GenericPiezoCLI import *
+from Thorlabs.MotionControl.Benchtop.PiezoCLI.PDXC2 import *
+from Thorlabs.MotionControl.Benchtop.Piezo.PDXC2 import *
+from Thorlabs.MotionControl.GenericPiezoCLI.Piezo import PiezoControlModeTypes
+
+class PDXC2(GenericDevice):
+
+    def __init__(self, serial: str = None, mode: int = 2) -> GenericDevice:
+        """Instantiates a PDXC2 object to control piezo stage.
+
+        :param str serial: Serial number
+        :param int mode: OpenLoop (1) or CloseLoop (2)
+        :return: PDXC2 object
+
+        """
+        self.device_prefix = InertiaStageController.DevicePrefix_PDXC2
+        super().__init__(serial=serial, device_prefix=self.device_prefix)
+        self.attempt_connection(mode)  
+        self.position = self.get_position()
+        self.control_mode = self.get_position_control_mode()
+        
+    def attempt_connection(self, mode: int) -> None:
+        """Attempt connection.
+
+        Generic connection attempt method. Will try to connect to specified
+        serial number after device lists have been built. Starts all relevant
+        routines as polling / command listeners ...
+
+        Args:
+            mode (int): Open or closed loop control mode. 1 for OpenLoop, 2 for CloseLoop.
+
+        """
+        try:
+            self.device = InertiaStageController.CreateInertiaStageController(self.serial)
+            self.device.Connect(self.serial)
+            timeout = 0
+            while not(self.device.IsSettingsInitialized()) and (timeout <= 10):
+                self.device.WaitForSettingsInitialized(500)
+                timeout += 1
+                if timeout == 10:
+                    raise RuntimeError('Settings failed to initialize.')
+                
+            self.device.StartPolling(250)
+            time.sleep(0.5)
+            self.device.EnableDevice()
+            time.sleep(0.1)
+            deviceInfo = [self.device.GetDeviceInfo()]
+            self.device_info = deviceInfo
+            self.device.EnableCommsListener
+            print("Success ! Connected to PDXC2 piezo:" +
+                  f" {self.device_info[0].Description}" + 
+                  f", S/N: {self.device_info[0].SerialNumber}"
+                  )
+
+            self.configuration = self.device.GetPDXC2Configuration(self.serial, DeviceConfiguration.DeviceSettingsUseOptionType.UseDeviceSettings)
```

Comment:
Need to change `DeviceConfiguration` to `PDXC2Configuration` since device manager is not imported anymore.